File: src/main/java/com/nugurang/graphql/resolver/VoteResolver.java
package com.nugurang.graphql.resolver;

import com.nugurang.dao.ArticleDao;
import com.nugurang.dao.UserDao;
import com.nugurang.dao.VoteTypeDao;
import com.nugurang.dto.ArticleDto;
import com.nugurang.dto.UserDto;
import com.nugurang.dto.VoteDto;
import com.nugurang.dto.VoteTypeDto;
import graphql.kickstart.tools.GraphQLResolver;
import org.springframework.stereotype.Service;

@Service
public class VoteResolver implements GraphQLResolver<VoteDto> {
    private final UserDao userDao;
    private final ArticleDao articledDao;
    private final VoteTypeDao voteTypeDao;

    public UserDto user(VoteDto voteDto) {
        return null;
    }

    public ArticleDto article(VoteDto voteDto) {
        return null;
    }

    public VoteTypeDto voteType(VoteDto voteDto) {
        return null;
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public VoteResolver(final UserDao userDao, final ArticleDao articledDao, final VoteTypeDao voteTypeDao) {
        this.userDao = userDao;
        this.articledDao = articledDao;
        this.voteTypeDao = voteTypeDao;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/resolver/VoteResolver.kt
package com.nugurang.graphql.resolver

import com.nugurang.dao.ArticleDao
import com.nugurang.dao.UserDao
import com.nugurang.dao.VoteTypeDao
import com.nugurang.dto.ArticleDto
import com.nugurang.dto.UserDto
import com.nugurang.dto.VoteDto
import com.nugurang.dto.VoteTypeDto
import graphql.kickstart.tools.GraphQLResolver
import org.springframework.stereotype.Service

@Service
class VoteResolver(
    private val userDao: UserDao,
    private val articledDao: ArticleDao,
    private val voteTypeDao: VoteTypeDao
) :

    GraphQLResolver<VoteDto?> {
    fun user(voteDto: VoteDto?): UserDto? {
        return null
    }

    fun article(voteDto: VoteDto?): ArticleDto? {
        return null
    }

    fun voteType(voteDto: VoteDto?): VoteTypeDto? {
        return null
    }
}